core/src/listener/listener.ts

```diff
@@ -215,30 +220,41 @@ function latestJob({
       throw e
     }
 
+    // happens only on localhost
     if (latestBlock < observedBlock) {
       logger.warn('latestBlock < observedBlock. Updating observed block to revert the condition.')
       observedBlock = Math.max(0, latestBlock - 1)
     }
 
-    const logPrefix = generateListenerLogPrefix(contractAddress, observedBlock, latestBlock)
+    let logPrefix = generateListenerLogPrefix(contractAddress, observedBlock, latestBlock)
```

Comment:
@Intizar-T How about not reusing the same variable for two different prefixes? This line can be moved inside of `try` block and kept `const`. The other one inside of `catch` block can be also defined as new variable with `const` modifier.